File: Alam's/agentra_user_frontend/lib/models/complaint.dart
class Complaint {
  final String id;
  final String userId;
  final String? bookingId;
  final String? agentId;
  final String? agentName;
  final String subject;
  final String description;
  final String status; // OPEN | IN_PROGRESS | RESOLVED
  final bool forwardedToAgent;
  final String? adminResponse;
  final String? agentResponse;
  final DateTime createdAt;

  Complaint({
    required this.id,
    required this.userId,
    this.bookingId,
    this.agentId,
    this.agentName,
    required this.subject,
    required this.description,
    required this.status,
    this.forwardedToAgent = false,
    this.adminResponse,
    this.agentResponse,
    required this.createdAt,
  });

  factory Complaint.fromJson(Map<String, dynamic> json) {
    final agentRaw = json['agentId'];
    return Complaint(
      id:               json['_id'] ?? json['id'] ?? '',
      userId:           json['userId'] is Map
                          ? (json['userId']['_id'] ?? '')
                          : (json['userId'] ?? ''),
      bookingId:        json['bookingId'] is Map
                          ? (json['bookingId']['_id'] ?? '')
                          : json['bookingId'],
      agentId:          agentRaw is Map ? (agentRaw['_id'] ?? '') : agentRaw,
      agentName:        agentRaw is Map
                          ? (agentRaw['businessName'] ?? agentRaw['fullName'])
                          : null,
      subject:          json['subject'] ?? '',
      description:      json['description'] ?? '',
      status:           json['status'] ?? 'OPEN',
      forwardedToAgent: json['forwardedToAgent'] ?? false,
      adminResponse:    json['adminResponse'] ?? json['ownerResponse'],
      agentResponse:    json['agentResponse'],
      createdAt:        json['createdAt'] != null
                          ? DateTime.parse(json['createdAt'])
                          : DateTime.now(),
    );
  }

  /// Human-readable status label
  String get statusLabel {
    switch (status.toUpperCase()) {
      case 'IN_PROGRESS': return 'In Progress';
      case 'RESOLVED':    return 'Resolved';
      default:            return 'Open';
    }
  }
}

File: Alam's/agentra-backend/src/models/Complaint.js
const mongoose = require('mongoose');

const complaintSchema = new mongoose.Schema({

  // ---------- Relations ----------
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },

  agentId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Agent'
  },

  bookingId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Booking'
  },

  // ---------- Complaint Content ----------
  subject: {
    type: String,
    required: true
  },

  description: {
    type: String,
    required: true
  },

  // ---------- Workflow ----------
  // OPEN       → user submitted, owner has not yet forwarded
  // IN_PROGRESS → owner forwarded to agent, agent working on it
  // RESOLVED   → agent marked as resolved (or owner resolved directly)
  status: {
    type: String,
    enum: ['OPEN', 'IN_PROGRESS', 'RESOLVED'],
    default: 'OPEN'
  },

  // Set to true when owner forwards the complaint to the agent
  forwardedToAgent: {
    type: Boolean,
    default: false
  },

  // owner's message when forwarding / responding
  ownerResponse: {
    type: String,
    default: ''
  },

  // Agent's resolution response
  agentResponse: {
    type: String,
    default: ''
  }

}, { timestamps: true });

module.exports = mongoose.model('Complaint', complaintSchema);
